hack/generate-jobs.py

```diff
@@ -0,0 +1,145 @@
+
+
+def generate(patterns: typing.List[str],
+             only_verify: bool, overwrite: bool) -> typing.List[str]:
+    """
+    Generate job configuration files.
+    Return list of errors.
+    """
+    errors = []
+    for pattern in patterns:
+        paths = glob.glob(pattern)
+        if not paths:
+            errors.append(f"No files found for pattern {pattern}")
+            continue
+        for fname in paths:
+            errs = generate_one(fname, only_verify, overwrite)
+            if errs:
+                errors.extend(errs)
+    return errors
+
+
+def generate_one(fname: str, only_verify: bool, overwrite: bool) -> typing.List[str]:
+    """
+    Generate job configuration files from one template.
+    Return list of errors.
+    """
+    config = configparser.ConfigParser()
+    config.read(fname)
+
+    template_name = config.get("DEFAULT", "template")
+    template_path = os.path.join(os.path.dirname(fname), template_name)
+    name = template_path.split(".")[0]
+    errors = []
+    with open(template_path, encoding="utf-8") as inp:
+        template = jinja2.Template(inp.read(), lstrip_blocks=True)
+        for kind in config.get("DEFAULT", "kinds").split(","):
+            tmp = tempfile.NamedTemporaryFile(
+                "w",
+                prefix=os.path.splitext(os.path.basename(__file__))[0],
+                delete=False,
+            )
+            with tmp:
+                header = (
+                    "# GENERATED FILE - DO NOT EDIT!\n#\n# "
+                    f"Instead, modify {template_name} and run `make generate-jobs`.\n"
+                )
+                for section in config.sections():
+                    tmp.write(
+                        template.render(
+                            config[section],
+                            kind=kind,
+                            job_name=f"{kind}-{section}",
+                            header=header,
+                        )
+                    )
+                    header = ""
+
+            out = f"{name}-{kind}.yaml"
+            if not os.path.exists(out):
+                if only_verify:
+                    os.unlink(tmp.name)
+                    errors.append(f"Can't verify content: {out} doesn't exist")
+                    continue
+            else:
+                equal = filecmp.cmp(out, tmp.name, shallow=False)
+                if only_verify:
+                    os.unlink(tmp.name)
+                    if not equal:
+                        errors.append(
+                            f"Generated content for {out} differs from existing"
+                        )
+                    continue
+                if equal:
+                    os.unlink(tmp.name)
+                    continue
+                if not overwrite:
+                    os.unlink(tmp.name)
+                    errors.append(
+                        f"Generated content for {out} differs from existing, "
+                        "use --owerwrite to update"
```

Comment:
There is no such parameter.

Reply:
@pohly 
> Can we perhaps use a pattern that is unique enough that we don't need to update it, for example config/jobs/**/*.generate-conf?

Done. It required a bit of refactoring as Python `glob.glob` doesn't understand this kind of pattern.